core/src/test/java/org/apache/iceberg/MetadataTableScanTestBase.java

```diff
@@ -59,6 +59,12 @@ protected Set<String> expectedManifestListPaths(
         .collect(Collectors.toSet());
   }
 
+  protected Set<String> actualManifestPaths(TableScan tableScan) {
```

Comment:
maybe we can call this entriesTableScan  .. as initially i was confused reading the method why this works.

Also we should probably combine at some point with actualManifestListPaths() method as its essentially the same thing.  Maybe in subsequent pr.

Reply:
yeah I think the type casting is actually not necessarily and the only difference is depends on scan type. I extracted common logic out to a separate method